src/lib/storage/index/segment_index_type.cpp

```diff
@@ -0,0 +1,23 @@
+#include "segment_index_type.hpp"
+
+
+namespace opossum {
+
+std::string segment_index_type_to_string(SegmentIndexType type) {
+  switch (type) {
```

Comment:
To get the enum as string, you can use `magic_enum::enum_name`, which we also use in other code locations.